### PR TITLE
Using the new tables_io functions for the parquet file iterator

### PR DESCRIPTION
**Pull request by joselotl**
## Code Quality
- [x] My code follows [the code style of this project](https://rail-hub.readthedocs.io/en/latest/source/contributing.html#naming-conventions)
- [x] I have written unit tests or justified all instances of `#pragma: no cover`; in the case of a bugfix, a new test that breaks as a result of the bug has been added
- [ ] My code contains relevant comments and necessary documentation for future maintainers; the change is reflected in applicable demos/tutorials (with output cleared!) and added/updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] Any breaking changes, described above, are accompanied by backwards compatibility and deprecation warnings

### CODE REVIEW

**Comment by joselotl** [comment]
@sidneymau If you have time, maybe you could try to run directly one estimator using directly your parquet files to see if this fix works.

File: src/rail/core/data.py [hunk 1]
```diff
@@ -367,6 +367,9 @@ class PqHandle(TableHandle):
 
     suffix = "pq"
 
+    @classmethod
+    def _size(cls, path, **kwargs):
+        return tables_io.io.getInputDataLengthPq(path, **kwargs)
```
Comment (hangqianjun) [inline src/rail/core/data.py:372]: Looks good. One question: it seems `tables_io.io.getInputDataLength` covers all hdf5 and parquet files, but why do we still need `tables_io.io.getInputDataLengthPq` specifically on line 372?

**Comment by eacharles** [comment]
I think that tables_io.io.getInputDataLength just decides function to call based on the type of data. I.e. it calls tables_io.io.getInputDataLengthPq or Hdf5, or whatever.On May 1, 2024, at 1:58 AM, hangqianjun ***@***.***> wrote:﻿
@hangqianjun commented on this pull request.



In src/rail/core/data.py:
> @@ -367,6 +367,9 @@ class PqHandle(TableHandle):
 
     suffix = "pq"
 
+    @classmethod
+    def _size(cls, path, **kwargs):
+        return tables_io.io.getInputDataLengthPq(path, **kwargs)
Looks good. One question: it seems tables_io.io.getInputDataLength covers all hdf5 and parquet files, but why do we still need tables_io.io.getInputDataLengthPq specifically on line 372?

**Comment by hangqianjun** [comment]
> I think that tables_io.io.getInputDataLength just decides function to call based on the type of data. I.e. it calls tables_io.io.getInputDataLengthPq or Hdf5, or whatever.

Yes exactly. I thought `getInputDataLength` already has the functionality so we can use it instead of `getInputDataLengthPq`. However it's no big issue, as the code still works! It just made me go look up the difference between the two when I saw this, and that's why I raised the question! Happy to approve the PR.

**Comment by eacharles** [comment]
So, to be clear, if you know what type of data you have you should probably call the appropriate function directly.   But if you aren’t sure, then you should call the generic function.-eOn May 1, 2024, at 8:48 AM, hangqianjun ***@***.***> wrote:﻿

I think that tables_io.io.getInputDataLength just decides function to call based on the type of data. I.e. it calls tables_io.io.getInputDataLengthPq or Hdf5, or whatever.

Yes exactly. I thought getInputDataLength already has the functionality so we can use it instead of getInputDataLengthPq. However it's no big issue, as the code still works! It just made me go look up the difference between the two when I saw this, and that's why I raised the question! Happy to approve the PR.

**Comment by hangqianjun** [comment]
> So, to be clear, if you know what type of data you have you should probably call the appropriate function directly.   But if you aren’t sure, then you should call the generic function.-e

I see! Thanks for the explanation!

**Review by hangqianjun — APPROVED** [review]

**Review by eacharles — APPROVED** [review]
Thanks Josue

**Pull request merged**